test/bin/mdx-pp/dune

```diff
@@ -0,0 +1,3 @@
+(executable
+ (name gen_dune_rules)
+ (libraries fmt gen_rule_helpers))
```

Comment:
Why not move this executable to where it's used, in `expect/` ? `expect/dune` uses a relative path (`../gen_dune_rule.exe`) to access this and it is not used in the `misc` directory.

Reply:
I try to maintain consistent structure between the various sub commands end to end tests, that is:
```
test/bin/mdx-test/
├── dune
├── gen_dune_rules.ml
├── expect
│   ├── ... # test case folders
│   ├── dune
│   └── dune.inc
├── failure
│   ├── ... # test case folders
│   ├── dune
│   └── dune.inc
└── misc
    └── ... # test case folders
```

And that even if we don't have error test cases for `ocaml-mdx pp` yet. Adding some remains simpler and I believe it makes it easier to navigate the end to end tests if each command folder has the same structure.

Reply:
I understood that. I was talking of:

```
test/bin/*/
├── expect
│   ├── ... # test case folders
│   ├── dune
│   ├── gen_dune_rules.ml
│   └── dune.inc
├── failure
│   ├── ... # test case folders
│   ├── dune
│   └── dune.inc
└── misc
    └── ... # test case folders
```

- `failure` and `misc` seems not to need `gen_dune_rules`
- Generated rules are very specific to "expect" tests

Reply:
`failure` does use `gen_dune_rules`

Reply:
You are right.